### PR TITLE
docs: Add MkDocs Material + mkdocstrings API reference site

### PR DESCRIPTION
**Pull request by xinghaohuang91**
## Summary

Adds a published API reference site for `twilio-agent-connect-aws`, mirroring the [twilio-agent-connect-python](https://github.com/twilio/twilio-agent-connect-python) docs pipeline:

- **MkDocs Material** theme + **mkdocstrings** for API pages generated from docstrings (Connectors, Server, Proxy, Tools)
- **mike** for version-aliased publishing; `.github/workflows/docs.yml` deploys to `gh-pages` on each GitHub release (or manually via `workflow_dispatch`), matching the Python repo's workflow line-for-line
- A `pull_request`-triggered `build` job (path-filtered to docs-relevant files) runs `mkdocs build --strict` without publishing, so broken config/snippets/mkdocstrings imports are caught in review instead of at release time — backported from review feedback on [twilio-agent-connect-microsoft#73](https://github.com/twilio/twilio-agent-connect-microsoft/pull/73)
- `docs/index.md` and `docs/deploy/**/README.md` are one-line snippet includes of the existing `README.md` and `deploy/*/README.md` guides — no content duplication, single source of truth
- `docs/api/proxy.md` and `docs/api/server.md` target individual modules rather than their packages, since `proxy/__init__.py` and `server/__init__.py` gate optional imports behind `try/except ModuleNotFoundError`, which mkdocstrings can't resolve statically
- Added an mkdocstrings `inventories` entry pointing at the published TAC Python docs, so TAC core types (`TAC`, `ConversationSession`, `SMSChannelConfig`, ...) resolve to cross-repo links automatically
- Fixed ~30 Markdown lists across `README.md`, `deploy/*/README.md`, and five docstrings that render fine on GitHub (which tolerates a list immediately after a text line) but collapsed into a single run-on paragraph under mkdocstrings' stricter Python-Markdown parser (needs a blank line before a list). Verified the two `Args:`-block fixes didn't change mkdocstrings' parsed parameter table by diffing rendered output before/after.
- Added a **Documentation** section to `README.md` linking the API reference, `CLAUDE.md`, and the Getting Started guide, matching the Python repo's README structure
- `make sync` stays `uv sync --all-extras --all-packages` — the docs dependency group is opt-in, installed on demand by `make docs`/`docs-serve` (which pass `--group docs` explicitly to `uv run`), so a plain dev setup doesn't pull in mkdocs and friends

**Site (already live from a manual first deploy, will auto-update from this PR's `docs.yml` on the next release):** https://twilio.github.io/twilio-agent-connect-aws/

### How to test

```bash
make sync
make docs-serve
```

Then open `http://127.0.0.1:8000/twilio-agent-connect-aws/` and check:
- Home page renders the README with working links
- API Reference → Connectors/Server/Proxy/Tools pages render class/function docs with no collapsed lists
- Deployment → each guide renders its architecture diagram (mermaid) and its `Key features:` bullets as an actual list
- `uv run --group docs mkdocs build --strict` exits clean (0 warnings)

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [x] Breaking change
- [x] Documentation update
- [ ] Refactoring

(Marking "Breaking change" only because `docs/deploy/*` page URLs are new paths — nothing in the package API changes. All five `src/tac_aws` edits are docstring whitespace only, confirmed via diff.)

## Checklist

- [x] Tests added/updated — n/a, no behavior change (`pytest` 82 passed, `mypy` 23 files clean, `ruff check`/`ruff format --check` clean)
- [x] Documentation updated — this PR is the documentation
- [x] Tested E2E — built and served locally, and verified all 10 published pages return 200 on the live `gh-pages` deploy

Co-Authored-By: Claude Sonnet 5 <noreply@anthropic.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div align="center">
   <div>
-    <img src="logo.svg" alt="Twilio Agent Connect AWS Logo" width="120" height="120">
+    <img src="https://raw.githubusercontent.com/twilio/twilio-agent-connect-aws/main/logo.svg" alt="Twilio Agent Connect AWS Logo" width="120" height="120">
```
[the remark this answers is not in the document]
Comment (xinghaohuang91) [reply README.md:3]: Confirmed — this repo's README is embedded into the mkdocs site via `--8<-- "README.md"` in `docs/index.md`, and the site is version-snapshotted with `mike`, so a historical version's page could technically point at a `main` that has since drifted.

That said, [twilio-agent-connect-python](https://github.com/twilio/twilio-agent-connect-python) has the identical pattern — every quick-link in its README (`LICENSE`, `CLAUDE.md`, `getting_started/README.md`, etc.) is hardcoded to `/main/` and published the same way via mike, unaddressed there too. We're treating this as a pre-existing upstream convention rather than something to solve in this PR, and the corner case (someone browsing an old pinned doc version and following a top-of-README badge link) is low-impact enough to accept for now. Leaving as-is; happy to revisit as a follow-up if it becomes a real pain point.

File: README.md [hunk 2]
```diff
@@ -14,16 +14,17 @@
   <div align="center">
     <a href="https://pypi.org/project/twilio-agent-connect-aws/"><img alt="PyPI" src="https://img.shields.io/pypi/v/twilio-agent-connect-aws.svg"/></a>
     <a href="https://github.com/twilio/twilio-agent-connect-aws"><img alt="Python SDK" src="https://img.shields.io/badge/Python-3.10+-3776AB.svg"/></a>
-    <a href="LICENSE"><img alt="License" src="https://img.shields.io/badge/license-MIT-green.svg"/></a>
+    <a href="https://github.com/twilio/twilio-agent-connect-aws/blob/main/LICENSE"><img alt="License" src="https://img.shields.io/badge/license-MIT-green.svg"/></a>
     <a href="https://github.com/twilio/twilio-agent-connect-aws/actions/workflows/ci.yml"><img alt="CI Status" src="https://github.com/twilio/twilio-agent-connect-aws/actions/workflows/ci.yml/badge.svg"/></a>
-    <a href="getting_started/examples/"><img alt="Getting Started" src="https://img.shields.io/badge/Getting%20Started-Examples-F22F46.svg"/></a>
+    <a href="https://github.com/twilio/twilio-agent-connect-aws/tree/main/getting_started/examples"><img alt="Getting Started" src="https://img.shields.io/badge/Getting%20Started-Examples-F22F46.svg"/></a>
```
[the remark this answers is not in the document]
Comment (xinghaohuang91) [reply README.md:19]: Same root cause as the [logo link comment](https://github.com/twilio/twilio-agent-connect-aws/pull/54#discussion_r3667622951) above — leaving as-is for the reasons noted there (pre-existing pattern in twilio-agent-connect-python too, low-impact corner case).

File: README.md [hunk 3]
```diff
@@ -14,16 +14,17 @@
...
   </div>
   
   <p>
     <a href="https://www.twilio.com/docs/platform/tac/overview">Documentation</a>
+    ◆ <a href="https://twilio.github.io/twilio-agent-connect-aws/">API Reference</a>
     ◆ <a href="https://github.com/twilio/twilio-agent-connect-python">Python SDK</a>
-    ◆ <a href="getting_started/examples/">Examples</a>
-    ◆ <a href="deploy/">Deployment</a>
+    ◆ <a href="https://github.com/twilio/twilio-agent-connect-aws/tree/main/getting_started/examples">Examples</a>
+    ◆ <a href="https://github.com/twilio/twilio-agent-connect-aws/tree/main/deploy">Deployment</a>
```
[the remark this answers is not in the document]
Comment (xinghaohuang91) [reply README.md:27]: Same root cause as the [logo link comment](https://github.com/twilio/twilio-agent-connect-aws/pull/54#discussion_r3667622951) above — leaving as-is for the reasons noted there (pre-existing pattern in twilio-agent-connect-python too, low-impact corner case).

File: Makefile [hunk 4]
```diff
@@ -46,6 +48,17 @@ check: lint type-check test
 pre-commit:
 	uv run pre-commit run --all-files
 
+docs:
+	@echo "Building documentation into site/..."
+	uv run --group docs mkdocs build
+
+docs-serve:
+	@echo "Serving docs at http://127.0.0.1:8000 (live reload)..."
+	uv run --group docs mkdocs serve
```
[the remark this answers is not in the document]
Comment (xinghaohuang91) [reply Makefile:57]: Fixed — `make docs` now runs `mkdocs build --strict`, matching the quality bar CLAUDE.md already tells contributors to use. `docs-serve` stays non-strict intentionally, since a live-reload dev loop shouldn't hard-fail on every warning mid-edit. Applied locally, will be in the next push.

File: .github/workflows/docs.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
+name: Docs
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version to publish (e.g. 1.1.0). Defaults to pyproject version."
+        required: false
```
[the remark this answers is not in the document]
Comment (xinghaohuang91) [reply .github/workflows/docs.yml:10]: Fixed — added a regex check right after `VERSION` is resolved (covers all three sources: release tag, manual dispatch input, and the pyproject.toml fallback) that rejects anything except `[A-Za-z0-9][A-Za-z0-9._-]*`, failing fast with `::error::` in the same style as the existing tag-validation step in deploy.yml. Tested against `../../etc`, `foo/bar`, `foo bar`, and empty string (all rejected) plus `1.1.0`, `v1.2.3`, `1.0.0-rc1` (all pass). Applied locally, will be in the next push.

File: .github/workflows/docs.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            VERSION="${GITHUB_REF_NAME#v}"
+          elif [ -n "$INPUT_VERSION" ]; then
+            VERSION="$INPUT_VERSION"
+          else
+            VERSION=$(python -c "
+          import tomllib
+          with open('pyproject.toml', 'rb') as f:
+              print(tomllib.load(f)['project']['version'])
+          ")
+          fi
```
[the remark this answers is not in the document]
Comment (xinghaohuang91) [reply .github/workflows/docs.yml:80]: Fixed — added a regex check right after `VERSION` is resolved (covers all three sources: release tag, manual dispatch input, and the pyproject.toml fallback) that rejects anything except `[A-Za-z0-9][A-Za-z0-9._-]*`, failing fast with `::error::` in the same style as the existing tag-validation step in deploy.yml. Tested against `../../etc`, `foo/bar`, `foo bar`, and empty string (all rejected) plus `1.1.0`, `v1.2.3`, `1.0.0-rc1` (all pass). Applied locally, will be in the next push.

File: .github/workflows/docs.yml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    if: github.repository_owner == 'twilio'
+    permissions:
+      contents: write # mike pushes the built site to the gh-pages branch
+      id-token: write
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          fetch-depth: 0 # mike needs full history of the gh-pages branch
+      - name: Authenticate with Artifactory
+        uses: twilio/sdk-actions/artifactory-oidc@23b656b843d2a3461cf38e4fd466c07a822d5fc0 # sdk-actions#7 (merged to main); repin to a release tag when one is cut
```
Comment (ryanrishi) [inline .github/workflows/docs.yml]: This has been merged, want to update this?

See https://github.com/twilio/twilio-agent-connect-aws/pull/53 for reference

Comment (xinghaohuang91) [reply .github/workflows/docs.yml]: sure! done!

**Review by ryanrishi — APPROVED** [review]
